libtenzir/builtins/operators/serve.cpp

```diff
@@ -862,12 +866,67 @@ class serve_operator final : public crtp_operator<serve_operator> {
 
 class plugin final : public virtual component_plugin,
                      public virtual rest_endpoint_plugin,
-                     public virtual operator_plugin<serve_operator> {
+                     public virtual operator_plugin<serve_operator>,
+                     public virtual aspect_plugin {
 public:
   auto component_name() const -> std::string override {
     return "serve-manager";
   }
 
+  auto aspect_name() const -> std::string override {
+    return "serves";
+  }
+
+  // this is for the aspect plugin
+  auto location() const -> operator_location override {
+    return operator_location::remote;
+  }
+
+  auto show(operator_control_plane& ctrl) const
+    -> generator<table_slice> override {
+    auto serve_manager = serve_manager_actor{};
+    auto blocking = caf::scoped_actor{ctrl.self().system()};
+    blocking
+      ->request(ctrl.node(), caf::infinite, atom::get_v, atom::type_v,
+                "serve-manager")
+      .receive(
+        [&](std::vector<caf::actor>& actors) {
+          TENZIR_ASSERT(actors.size() == 1);
+          serve_manager
+            = caf::actor_cast<serve_manager_actor>(std::move(actors[0]));
+        },
+        [&](const caf::error& err) { //
+          ctrl.abort(caf::make_error(
+            ec::logic_error,
+            fmt::format("failed to find serve-manager: {}", err)));
+        });
+    co_yield {};
+    auto serves = list{};
+    blocking
+      ->request(serve_manager, caf::infinite, atom::status_v,
+                status_verbosity::debug, duration{std::chrono::seconds{10}})
```

Comment:
Is this timeout ever a problem?

Do we have a tunable constant for this?

Reply:
Process-internal requests with timeouts shouldn't be a problem. If this takes a long time to get answered, it can still get cancelled at any time if the operator gets shut down from the outside, e.g., because the pipeline TTL expired for deployed pipelines or because the user pressed C-c on the command-line.